docs/source/cpp/api/builder.rst

```diff
@@ -34,6 +34,9 @@ Primitive
 .. doxygenclass:: arrow::BooleanBuilder
    :members:
 
+.. doxygenclass:: arrow::HalfFloatBuilder
```

Comment:
This shouldn't be necessary if `HalfFloatBuilder` is added to the `numeric-builders` group.